webui/apps/eidolon-ui2/app/eidolon-apps/dev-tool/components/start-program-dialog.tsx

```diff
@@ -5,7 +5,7 @@ import * as React from "react";
 import {usePathname, useRouter} from "next/navigation";
 import {ChooseAgentForm, getAppPathFromPath} from "@eidolon-ai/components/client";
 
-export function StartProgramDialog({open, onClose, machineUrl}: { machineUrl: string, open: boolean, onClose: (wasCanceled: boolean) => void }) {
+export function StartProgramDialog({open, onClose, machineUrl, defaultAgent}: { machineUrl: string, open: boolean, defaultAgent: string | undefined, onClose: (wasCanceled: boolean) => void }) {
```

Comment:
Would it make sense to use `defaultAgent?: string` instead? Wondering if there's a use case for `undefined` here.

Reply:
No, you are right, this is better. I don't know typescript.